etcdserver/api/v3rpc/watch.go

```diff
@@ -56,9 +57,14 @@ var (
 )
 
 func GetProgressReportInterval() time.Duration {
+	// add rand(1/10*progressReportInterval) as jitter so that etcdserver will not
+	// send progress notifications to watchers around the same time even when watchers
+	// are created around the same time (which is common when a client restarts itself).
+	jitter := time.Duration(rand.Int63n(int64(progressReportInterval) / 10))
```

Comment:
Should `int64(progressReportInterval) / 10` be done within the Rlock?

Reply:
@cw9 i still keep the rand func out of the lock for good locking practices to minimize the critical section since rand can be expensive.